// @react-navigation/material-top-tabs/lib/module/views/MaterialTopTabBar.js
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import { Text } from '@react-navigation/elements';
import { useLinkBuilder, useLocale, useTheme } from '@react-navigation/native';
import Color from 'color';
import * as React from 'react';
import { StyleSheet } from 'react-native';
import { TabBar, TabBarIndicator } from 'react-native-tab-view';
const MaterialLabel = ({
  color,
  labelText,
  labelStyle,
  allowScaling
}) => {
  return /*#__PURE__*/React.createElement(Text, {
    style: [{
      color
    }, styles.label, labelStyle],
    allowFontScaling: allowScaling
  }, labelText);
};
const renderLabel = props => {
  return /*#__PURE__*/React.createElement(MaterialLabel, props);
};
export function MaterialTopTabBar({
  state,
  navigation,
  descriptors,
  ...rest
}) {
  const {
    colors
  } = useTheme();
  const {
    direction
  } = useLocale();
  const {
    buildHref
  } = useLinkBuilder();
  const focusedOptions = descriptors[state.routes[state.index].key].options;
  const activeColor = focusedOptions.tabBarActiveTintColor ?? colors.text;
  const inactiveColor = focusedOptions.tabBarInactiveTintColor ?? Color(activeColor).alpha(0.5).rgb().string();
  const tabBarOptions = Object.fromEntries(state.routes.map(route => {
    const {
      options
    } = descriptors[route.key];
    return [route.key, {
      href: buildHref(route.name, route.params),
      testID: options.tabBarButtonTestID,
      accessibilityLabel: options.tabBarAccessibilityLabel,
      badge: options.tabBarBadge,
      icon: options.tabBarShowIcon === false ? undefined : options.tabBarIcon,
      label: options.tabBarShowLabel === false ? undefined : renderLabel,
      labelAllowFontScaling: options.tabBarAllowFontScaling,
      labelStyle: options.tabBarLabelStyle,
      labelText: options.tabBarShowLabel === false ? undefined : options.title !== undefined ? options.title : route.name
    }];
  }));
  return /*#__PURE__*/React.createElement(TabBar, _extends({}, rest, {
    navigationState: state,
    options: tabBarOptions,
    direction: direction,
    scrollEnabled: focusedOptions.tabBarScrollEnabled,
    bounces: focusedOptions.tabBarBounces,
    activeColor: activeColor,
    inactiveColor: inactiveColor,
    pressColor: focusedOptions.tabBarPressColor,
    pressOpacity: focusedOptions.tabBarPressOpacity,
    tabStyle: focusedOptions.tabBarItemStyle,
    indicatorStyle: [{
      backgroundColor: colors.primary
    }, focusedOptions.tabBarIndicatorStyle],
    gap: focusedOptions.tabBarGap,
    android_ripple: focusedOptions.tabBarAndroidRipple,
    indicatorContainerStyle: focusedOptions.tabBarIndicatorContainerStyle,
    contentContainerStyle: focusedOptions.tabBarContentContainerStyle,
    style: [{
      backgroundColor: colors.card
    }, focusedOptions.tabBarStyle],
    onTabPress: ({
      route,
      preventDefault
    }) => {
      const event = navigation.emit({
        type: 'tabPress',
        target: route.key,
        canPreventDefault: true
      });
      if (event.defaultPrevented) {
        preventDefault();
      }
    },
    onTabLongPress: ({
      route
    }) => navigation.emit({
      type: 'tabLongPress',
      target: route.key
    }),
    renderIndicator: ({
      navigationState: state,
      ...rest
    }) => {
      return focusedOptions.tabBarIndicator ? focusedOptions.tabBarIndicator({
        state: state,
        ...rest
      }) : /*#__PURE__*/React.createElement(TabBarIndicator, _extends({
        navigationState: state
      }, rest));
    }
  }));
}
const styles = StyleSheet.create({
  label: {
    textAlign: 'center',
    fontSize: 14,
    margin: 4,
    backgroundColor: 'transparent'
  }
});
